File: modules/agent/src/mocks/repositories/orderRepo.ts
import { readFileSync } from 'fs';
import { dirname, join } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));
const DATA_PATH = join(__dirname, '..', 'data');

export interface OrderItem {
  sku: string;
  qty: number;
  unit_price: number;
  location: string;
}

export interface Order {
  id: string;
  customer_id: string;
  status: string;
  created_at: string;
  shipped_at: string | null;
  items: OrderItem[];
  total: number;
}

let orders: Order[] = [];
let validStatuses: string[] = [];

const loadData = (): void => {
  if (orders.length > 0) return;

  const data = JSON.parse(readFileSync(join(DATA_PATH, 'orders.json'), 'utf-8'));
  orders = data.orders;
  validStatuses = data.statuses;
};

export const getAllOrders = (): Order[] => {
  loadData();
  return orders;
};

export const findOrderById = (orderId: string): Order | undefined => {
  loadData();
  return orders.find((o) => o.id === orderId);
};

export const appendOrder = (order: Order): Order => {
  loadData();
  orders.push(order);
  return order;
};

export const getValidStatuses = (): string[] => {
  loadData();
  return validStatuses;
};

export const getOrderCount = (): number => {
  loadData();
  return orders.length;
};

File: modules/agent_deprecated/src/agent/mocks/repositories/order_repo.py
import json
from pathlib import Path

DATA_PATH = Path(__file__).parent.parent / "data"

_orders: list[dict] = []
_valid_statuses: list[str] = []


def _load_data():
    global _orders, _valid_statuses
    if _orders:
        return
    with open(DATA_PATH / "orders.json") as f:
        data = json.load(f)
    _orders = data["orders"]
    _valid_statuses = data["statuses"]


def get_all_orders() -> list[dict]:
    _load_data()
    return _orders


def find_order_by_id(order_id: str) -> dict | None:
    _load_data()
    return next((o for o in _orders if o["id"] == order_id), None)


def append_order(order: dict) -> dict:
    _load_data()
    _orders.append(order)
    return order


def get_valid_statuses() -> list[str]:
    _load_data()
    return _valid_statuses


def get_order_count() -> int:
    _load_data()
    return len(_orders)
